launcher/FileSystem.h

```diff
@@ -358,6 +369,14 @@ QString getDesktopDir();
 // Get the Directory representing the User's Applications directory
 QString getApplicationsDir();
 
+#if defined(Q_OS_LINUX)
+// Check if we're running inside an AppImage
+bool runningInAppImage(QCoreApplication& app);
+
+// Check if we're running inside a self-contained Linux bundle
+bool runningSelfContained(QCoreApplication& app);
+#endif
```

Comment:
DesktopServices.h is where we have other similar things, it's a little weird to suddenly start putting them in FileSystem